src/solvers/smt2/smt2_solver.cpp

```diff
@@ -39,6 +38,13 @@ class smt2_solvert:public smt2_parsert
   void expand_function_applications(exprt &);
 
   std::set<irep_idt> constants_done;
+
+  enum
+  {
+    NOT_SOLVED,
+    SAT,
+    UNSAT
+  } status;
```

Comment:
Why do we need yet another status enum? We have one for decision procedures, one in prop.h - wouldn't the latter be the right one?